vllm/worker/enc_dec_model_runner.py

```diff
@@ -128,28 +116,11 @@ def raise_backend_err():
         maybe_global_forced_backend = get_global_forced_attn_backend()
         is_forced_by_global = maybe_global_forced_backend is not None
         is_forced_by_env_var = maybe_env_var_forced_backend is not None
-
-        if not (is_forced_by_global or is_forced_by_env_var) \
-            and self._is_xformers_only_encoder_decoder_model(model):
-            # The user has not already specified an attention backend
-            # override
-            logger.info(
-                "Encoder-Decoder Model Architecture %s requires XFormers "
-                "backend; overriding backend auto-selection and "
-                "forcing XFormers.", get_architecture_class_name(model))
-            global_force_attn_backend(_Backend.XFORMERS)
-        elif is_forced_by_global:
-            # Backend override enforced by global variable takes
-            # precedence over vLLM backend environment variable.
-            if maybe_global_forced_backend not in\
-                 [_Backend.XFORMERS, _Backend.FLASH_ATTN]:
-                raise_backend_err()
-        elif is_forced_by_env_var:
-            # Backend override enforced by vLLM backend
-            # environment variable
-            if maybe_env_var_forced_backend not in\
-                 [_Backend.XFORMERS, _Backend.FLASH_ATTN]:
-                raise_backend_err()
+        if (is_forced_by_global and maybe_global_forced_backend \
+            not in [_Backend.XFORMERS, _Backend.FLASH_ATTN]) or \
+            (is_forced_by_env_var and maybe_env_var_forced_backend \
+                not in [_Backend.XFORMERS, _Backend.FLASH_ATTN]):
+            raise_backend_err()
```

Comment:
I suggest to revert it to the code before #9559
```
if is_forced_by_global:
    # Backend override enforced by global variable takes
    # precedence over vLLM backend environment variable.
    if maybe_global_forced_backend != _Backend.XFORMERS:
        raise_backend_err()
elif is_forced_by_env_var:
    # Backend override enforced by vLLM backend
    # environment variable
    if maybe_env_var_forced_backend != _Backend.XFORMERS:
        raise_backend_err()
```

Reply:
Done. I combined them because of a lint. Reverted an added the ignore.